shipping/uc_usps/uc_usps.module

```diff
@@ -579,6 +645,433 @@ function uc_usps_intl_rate_request($packages, $origin, $destination) {
+      // combine the results.
+
+      $errors = array();
+      $rate_info = uc_usps_newapi_package_quote($package, $origin, $destination, $method, $errors);
+      $num_packages++;
+      foreach ($rate_info as $key => $info) {
+        $rate = $info['totalBasePrice'];
+        if (!isset($services[$key])) {
+          $services[$key] = array(
+            'rate' => uc_usps_rate_markup((string) $rate),
+            'label' => $info['description'],
+            'option_label' => theme('uc_usps_option_label', array(
+              'service' => $info['description'],
+              'packages' => $packages,
+            )),
+            'count' => 1,
+          );
+        }
+        else {
+          $services[$key]['rate'] += $rate;
+          $services[$key]['count']++;
+        }
+      }
+      if (!empty($errors)) {
+        foreach ($errors as $error) {
+          $debug_data['error'][] = $error['code'] . ' ' . $error['message'];
+        }
+      }
+    }
+  }
+
+  // USPS module ships all packages via the same rate, so filter the list to
+  // include only those services that were quoted for all packages.
+  foreach ($services as $key => $service) {
+    if ($service['count'] != $num_packages) {
+      unset($services[$key]);
+    }
+    else {
+      unset($services[$key]['count']);
+    }
+  }
+
+  // Filter the list to include only those services that were checked in the
+  // settings.
+  $method_services = array(
+    'usps_env' => 'uc_usps_newapi_dom_env_services',
+    'usps' => 'uc_usps_newapi_dom_parcel_services',
+    'usps_intl_env' => 'uc_usps_newapi_intl_env_services',
+    'usps_intl' => 'uc_usps_newapi_intl_parcel_services',
+  )[$method['id']];
+  $usps_services = array_filter(config_get('uc_usps.settings', $method_services));
+  foreach ($services as $service => $quote) {
+    if (!in_array($service, $usps_services)) {
+      unset($services[$service]);
+    }
+  }
+
+  // Sort by price.
+  uasort($services, 'uc_quote_price_sort');
+
+  // Merge debug data into $services.  This is necessary because $debug_data is
+  // not sortable by a 'rate' key, so it has to be kept separate from the
+  // $services data until this point.
+  if (isset($debug_data['debug']) ||
+      (isset($debug_data['error']) && count($debug_data['error']))) {
+    $services['data'] = $debug_data;
+  }
+
+  return $services;
+}
+
+/**
+ * Get a USPS quote on a single package using the new API.
+ *
+ * @param object $package - the package to be quoted.
+ * @param UcAddress $origin - its origin address.
+ * @param object $destination - its destination.
+ * @param array $method - the quote method.
+ * @param array $errors - a list that will be augmented with diagnostic
+ * information if there are any errors.
+ *
+ * @return array $rate_info - an array of rate information for this package.
+ *  The fields of each row are those returned by USPS and vary by mail class.
+ *  Each rate has a key so that multi-package quotes can be combined by key.
```

Comment:
```suggestion
 * @return array
 *   An array of rate information for this package.
 *   The fields of each row are those returned by USPS and vary by mail class.
 *   Each rate has a key so that multi-package quotes can be combined by key.
```